### PR TITLE
Add implementation of Gerrit human-comments details

### PR DESCRIPTION
**Pull request by gferon**
The idea is to properly relay human comments, while still excluding long "bot" comments.

Inline comments should also be supported.

### CODE REVIEW

File: src/gerrit.rs [hunk 1]
```diff
@@ -299,3 +310,40 @@ mod test {
...
+    );
+    ssh_channel.exec(&query).unwrap();
+
+    let buf_channel = BufReader::new(ssh_channel);
+    let line = buf_channel.lines().next();
+
+    // event from our channel cannot fail
+    let json: String = line.unwrap().ok().unwrap();
+    let res: Result<Change, _> = serde_json::from_str(&json);
+    println!("[D] {:?} for json: {}", res, json);
```
Comment (fbenkstein) [inline src/gerrit.rs]: debug!()?

Comment (gferon) [reply src/gerrit.rs]: Yes.

**Comment by gferon** [comment]
Ok, I managed to do something that works, with all the error handling that goes well. I think I made a mess with inconsistent borrows and other stuff - so please give me some feedback.

File: src/args.rs [hunk 2]
```diff
@@ -86,6 +85,6 @@ pub fn parse_config(path: PathBuf) -> Config {
         eprintln!("Could not parse config file: {}", e);
         ::std::process::exit(2)
     });
-    println!("{:?}", config);
+    debug!("{:?}", config);
```
Comment (boxdot) [inline src/args.rs]: If you want to do this, you need to move the initialization of the stderrlog before config parsing.

File: src/main.rs [hunk 3]
```diff
@@ -157,14 +168,15 @@ fn main() {
...
                         let gerrit_config = config.gerrit.clone();
-                        let gerrit_hostname = config.gerrit.hostname.clone();
-                        let gerrit_change = match gerrit::query(
-                            &gerrit_config.hostname,
-                            gerrit_config.port,
-                            &gerrit_config.username,
-                            &gerrit_config.priv_key_path,
-                            &change_id,
-                        ) {
+                        let gerrit_host = config.gerrit.host.clone();
```
Comment (boxdot) [inline src/main.rs]: Hmm. I would prefer to move this logic to a different thread, and communicate through a pipe.

Comment (fbenkstein) [reply src/main.rs]: I can give that a shot.

Comment (fbenkstein) [reply src/main.rs]: Have a look here, please:  https://github.com/fbenkstein/gerritbot-rs/commit/f7fd0d3a278b569786b0004935b93a050bf54c82

Comment (gferon) [reply src/main.rs]: I like the approach, you should push directly to this PR!

Comment (boxdot) [reply src/main.rs]: Thanks @fbenkstein. I used your idea but without using unsafe. I just introduced another function: reconnect.

**Comment by boxdot** [comment]
Still TODO:
* [ ] Inline comments need to be fetched only when the main message
  contains a number and `comments` word.
* [ ] Add review comment also to the bot notification.
* [ ] Deduplicate code in `connect` and `reconnect`.
* [ ] Write tests.

File: src/args.rs (hunk 2 above)
Comment (gferon) [reply src/args.rs]: done in additional commit.

**Comment by boxdot** [comment]
* Deduplication of the code is out of scope for now.
* To do proper testing of this feature we need proper channel replacement tests.

**Pull request merged**